asset/asset.go

```diff
@@ -980,6 +983,41 @@ func (g *GroupKeyRevealV0) Encode(w io.Writer) error {
 	return nil
 }
 
+// Decode decodes the group key reveal from the reader.
+func (g *GroupKeyRevealV0) Decode(r io.Reader, buf *[8]byte, l uint64) error {
+	// Verify that the group key reveal is not excessively long. This check
+	// is essential to prevent misinterpreting V1 and later group key
+	// reveals as V0.
+	if l > btcec.PubKeyBytesLenCompressed+sha256.Size {
+		return tlv.ErrRecordTooLarge
+	}
+
+	if l < btcec.PubKeyBytesLenCompressed {
```

Comment:
Stye nit: could be combined above into a switch statement.

Reply:
This commit is a code move/copy. I was trying not to make any code changes other than copying to maximise clarity and minimise the review burden. I agree that switch statement would be nice! Perhaps after this I can do some refactoring.